processor/process.go

```diff
@@ -0,0 +1,45 @@
+package processor
+
+import (
+	"crypto/aes"
+	"crypto/cipher"
+	"encoding/base64"
+	"fmt"
+	"serial-data-decryptor/models"
+)
+
+func Process(msg models.Data) ([]byte, error) {
+	return Decrypt(msg)
+}
+
+func Decrypt(msg models.Data) ([]byte, error) {
+	var key = []byte{0x00, 0x01, 0x02, 0x03, 0x04, 0x05, 0x06, 0x07, 0x08, 0x09, 0x0a, 0x0b, 0x0c, 0x0d, 0x0e, 0x0f}
+
+	iv, err := base64.StdEncoding.DecodeString(fmt.Sprint(msg["data"]))
+	if err != nil {
+		return nil, err
+	}
+	ct, err := base64.StdEncoding.DecodeString(fmt.Sprint(msg["iv"]))
```

Comment:
shouldn't it be the other way around?

Reply:
like?

Reply:
you read `msg["data"]` and assign it to `iv`. then you read `msg["iv"]` and assign it to `ct`. this seems to be mixed up

Reply:
ooh, my bad, fixed it, and also added validation to avoid Panic by aesgcm.Open